public/components/common/hocs/error-boundary/with-error-boundary.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Wazuh app - React HOCs to manage user authorization requirements
+ * Copyright (C) 2015-2021 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+
+import React, { Component } from 'react';
+import log from 'loglevel';
+import { EuiEmptyPrompt } from '@elastic/eui';
+
+export default class ErrorBoundary extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { error: null, errorInfo: null };
+  }
+
+  componentDidCatch = (error, errorInfo) => catchFunc(error, errorInfo, this);
+
+  render() {
+    // @ts-ignore
+    if (this.state.errorInfo) {
+      return <HandleError props={this} />;
+    }
+    return this.props.children;
+  }
+}
+
+export const withErrorBoundary = (WrappedComponent) => (props) => (
+  <ErrorBoundary>
+    <WrappedComponent {...props} />
+  </ErrorBoundary>
+);
+
+const catchFunc = (error, errorInfo, ctx) => {
+  ctx.setState({
+    error: error,
+    errorInfo: errorInfo,
+  });
+
+  log.error({ error, errorInfo });
+};
+
+const ErrorComponent = (props: { ctx: any }) => {
```

Comment:
it would be nice if was a child component for future enhancements

Reply:
Great!